src/skmultiflow/data/data_generator.py

```diff
@@ -0,0 +1,74 @@
+import pandas as pd
+import numpy as np
+from skmultiflow.data.base_generator import BaseGenerator
+
+# TODO : doc for all methods
+
+class DataGenerator(BaseGenerator):
+    # TODO : inherit from pandas.DataFrame should make things easier
+    """ DataGenerator
+
+    A generator constructed from the entries of a static dataset (numpy array or pandas
+    DataFrame).
+
+    The batch generator is able to provide, as requested, a number of samples, in
+    a way that old samples cannot be accessed in a later time. This is done
+    so that a stream context can be correctly simulated.
+
+    Parameters
+    ----------
+    data: np.ndarray or pd.DataFrame
+        The features' columns and targets' columns or the feature columns
+        only if they are passed separately.
+    """
+
+    def __init__(self, data, y=None, return_np=False):
+        BaseGenerator.__init__(self)
+
+        if not isinstance(data, pd.DataFrame):
+            data = pd.DataFrame(data=data)
+
+        self.data = data
+        self.sample_idx = 0
+
+        self.return_np = return_np
+
+    def next_sample(self, batch_size=1):
+        sample = self.data.iloc[self.sample_idx:self.sample_idx + batch_size]
+        self.sample_idx += batch_size
+        if not sample.empty:
+            return sample.values if self.return_np else sample
+        else:
+            return (None, None)
+
+    def last_sample(self):
+        return self.next_sample()
```

Comment:
Is this intentional? I would not expect last sample and next sample to mean the same thing.

Reply:
@martindurant actually the way it is implemented in mutlfilow is that generator (formerly streams) keep the last "emitted" batch as an internal attributes. 
I can change this piece to fit more precisely to the definition

Reply:
So long as it's documented :)

Reply:
This is indeed misleading. Here is the current implementation in the `Stream` class:
``` python
    def last_sample(self):
        """ Retrieves last `batch_size` samples in the stream.

        Returns
        -------
        tuple or tuple list
            A numpy.ndarray of shape (batch_size, n_features) and an array-like of shape
            (batch_size, n_targets), representing the next batch_size samples.

        """
        return self.current_sample_x, self.current_sample_y
```

**Note:** The docstring should read "...representing the **last** batch_size samples."